src/core/qgstiledownloadmanager.cpp

```diff
@@ -50,21 +50,36 @@ void QgsTileDownloadManagerWorker::queueUpdated()
 
   if ( mManager->mShuttingDown )
   {
+    // here we HAVE to build up a list of replies from the queue before do anything
+    // with them. Otherwise we can hit the situation where aborting the replies
+    // triggers immediately their removal from the queue, and we'll be modifying
+    // mQueue elsewhere while still trying to iterate over it here => crash
+    QList< QNetworkReply * > replies;
```

Comment:
Out of interest, why not an std:: container here?

Reply:
good point, fixed